ui/app/services/breadcrumbs.js

```diff
@@ -1,26 +1,32 @@
 import { getOwner } from '@ember/application';
-import Service, { inject as service } from '@ember/service';
 import { computed } from '@ember/object';
-import classic from 'ember-classic-decorator';
+import Service, { inject as service } from '@ember/service';
 
-@classic
 export default class BreadcrumbsService extends Service {
   @service router;
 
-  // currentURL is only used to listen to all transitions.
-  // currentRouteName has all information necessary to compute breadcrumbs,
-  // but it doesn't change when a transition to the same route with a different
-  // model occurs.
-  @computed('router.{currentURL,currentRouteName}')
-  get breadcrumbs() {
-    const owner = getOwner(this);
-    const allRoutes = (this.get('router.currentRouteName') || '')
+  generatePathHierarchy = (routeName = '') => {
+    return routeName
       .split('.')
       .without('')
-      .map((segment, index, allSegments) => allSegments.slice(0, index + 1).join('.'));
+      .map((_, index, allSegments) => allSegments.slice(0, index + 1).join('.'));
+  };
+  generateCrumb = route => {
+    if (route.breadcrumbs) {
+      const areBreadcrumbsAFunction = typeof route.breadcrumbs === 'function';
+
+      if (areBreadcrumbsAFunction) {
+        return route.breadcrumbs(route.get('controller.model')) ?? [];
+      } else {
+        return route.breadcrumbs;
+      }
+    } else {
+      return [];
+    }
+  };
 
-    let crumbs = [];
-    allRoutes.forEach(routeName => {
+  generateBreadcrumbs = (fullRouteNamesOfPathHierarchy, owner) =>
+    fullRouteNamesOfPathHierarchy.reduce((crumbs, routeName) => {
       const route = owner.lookup(`route:${routeName}`);
 
       // Routes can reset the breadcrumb trail to start anew even
```

Comment:
It's not in the PR, so I can't highlight it exactly, but just out of curiosity, do you know where `resetBreadcrumbs` is defined? I couldn't find it anywhere else in the code 🤔